File: src/main/java/io/accelerate/solutions/CHK/CheckoutSolution.java
package io.accelerate.solutions.CHK;

import io.accelerate.runner.SolutionNotImplementedException;

public class CheckoutSolution {
    public Integer checkout(String skus) {
        if(skus.length()<1){
            return -1;
        }
        Integer count = 0;
        Integer[] skuRepeated = new Integer[4];
        char [] skusArr = skus.toCharArray();
        for(char sku : skusArr) {
            switch (sku) {
                case 'A':
                    skuRepeated[0]++;
                    break;
                case 'B':
                    skuRepeated[1]++;
                    break;
                case 'C':
                    skuRepeated[2]++;
                    break;
                case 'D':
                    skuRepeated[3]++;
                    break;
                default:
                    return -1;
            }
        }

        for(int i=0;i<skuRepeated.length;i++){
            
        }


        return count;
    }
}

File: src/BeFaster.App/Solutions/CHK/CheckoutSolution.cs
using BeFaster.Runner.Exceptions;

namespace BeFaster.App.Solutions.CHK
{
    public class CheckoutSolution
    {
        public int Checkout(string? skus)
        {
            var info = new Dictionary<char, int>
            {
                { 'A', 50 },
                { 'B', 30 },
                { 'C', 20 },
                { 'D', 15 },
                { 'E', 40 }
            };

            var offers = new Dictionary<char, List<int[]>> {

                { 'A', [ new int[] { 3, 120 }, new int[] { 5, 200 } ] },
                { 'B', [ new int[] { 2, 45 } ] },
            };

            if (skus != null)
            {
                var counts = new Dictionary<char, int>();
                foreach (var key in skus.ToCharArray())
                {
                    var item = counts.GetValueOrDefault(key);
                    if (item == 0)
                    {
                        counts.Add(key, 1);
                    }
                    else
                    {
                        counts[key]++;
                    }
                }
                var total = 0;
                foreach (var entry in counts)
                {
                    var item = entry.Key;
                    var count = entry.Value;
                    var price = info.GetValueOrDefault(item);

                    if (offers.TryGetValue(item, out var discount))
                    {
                        
                    }

                    total += count * price;
                }
                return total;
            }
            else
            {
                return -1;
            }
        }
    }
}







